ui/plugin-system/src/model/variables.ts

```diff
@@ -18,29 +18,30 @@ import { usePlugin } from '../components/PluginLoadingBoundary';
  * Plugin for handling custom VariableDefinitions.
  */
 export interface VariablePlugin<Options extends JsonObject = JsonObject> {
-  useVariableOptions: UseVariableOptionsHook<Options>;
+  getVariableOptions: GetVariableOptions<Options>;
 }
 
 /**
  * Plugin hook responsible for getting the options of a custom variable
  * definition.
  */
-export type UseVariableOptionsHook<Options extends JsonObject = JsonObject> = (
+export type GetVariableOptions<Options extends JsonObject = JsonObject> = (
```

Comment:
Also, I didn't pass any context yet since I didn't need it for the plugins yet. But I imagine, i'll inject the current template variable states (needed for chaining variables) and time range (so template variables that use time can be recomputed)